src/other/sample.jl

```diff
@@ -0,0 +1,28 @@
+import StatsBase: sample
+
+function sample(df::DataFrame)
+    sample(df, 1)
+end
+
+"""
+    sample(df[, N])
+
+Returns a (random) sample of a DataFrame
+
+# Example
+```
+julia> using RDatasets
+julia> iris = dataset("datasets", "iris")
+julia> sample(iris, 5)
+5×5 DataFrames.DataFrame
+│ Row │ SepalLength │ SepalWidth │ PetalLength │ PetalWidth │ Species      │
+│ 1   │ 5.6         │ 2.9        │ 3.6         │ 1.3        │ "versicolor" │
+│ 2   │ 6.4         │ 3.1        │ 5.5         │ 1.8        │ "virginica"  │
+│ 3   │ 5.1         │ 3.8        │ 1.6         │ 0.2        │ "setosa"     │
+│ 4   │ 5.5         │ 2.4        │ 3.7         │ 1.0        │ "versicolor" │
+│ 5   │ 5.1         │ 3.8        │ 1.6         │ 0.2        │ "setosa"     │
+```
+"""
+function sample(df::DataFrame, N::Integer)
```

Comment:
The kw arg `replace` is missing.

Would be great to add to new kw args: `stratum` and `orderby` to allow [stratified](https://en.wikipedia.org/wiki/Stratified_sampling) and [systematic](https://en.wikipedia.org/wiki/Systematic_sampling) sampling respectively. That would be easy to implement using the `by` and `sort` functions i.e.:

```julia
function sample(df::DataFrame, N::Integer; stratum=nothing, orderby=...)
...
if stratum !== nothing
    by(df, stratum) do subset
        if sort !== nothing
            sorted = sort(subset, cols=orderby)
            ...sample...
...
``` 
If that's a complicated API, two new functions `systematic_sample` and `stratified_sample` can be added instead of the kw args.